### DISCUSSION TITLE
Configuration Error packageRules[0]: Each packageRule must contain at least one non-match* or non-exclude* field

### DISCUSSION THREAD

**Discussion by codewode** [renovatebot/renovate#13580] [Request Help] [answered]
I get the following warning

```
"warnings": [
         {
           "topic": "Configuration Error",
           "message": "packageRules[0]: Each packageRule must contain at least one non-match* or non-exclude* field. Rule: {\"matchUpdateTypes\":[\"minor\",\"patch\",\"pin\",\"digest\"]}"
         }
       ]

```

here is my config
```
{
  "rangeStrategy": "pin",
  "baseBranches": ["dependency-updates"],
  "packageRules": [
    { 
      "updateTypes": ["minor", "patch", "pin", "digest"]
    }
  ],
  "followTag": ["renovate"]
}
```

I did not get any PR.

**Comment by rarkins** [comment, marked as answer]
Your packageRule is not correct, so it's right that you get an error. What are you hoping to achieve with it?

**Reply by codewode** [reply]
but not sure why the other config works, the only difference is I removed "followTag" and used "main" branch instead.

```
{
  "platform": "azure",
  "rangeStrategy": "pin",
  "baseBranches": ["main"],
  "packageRules": [
    {
      "updateTypes": ["minor", "patch", "pin", "digest"]
    }
  ]
}
```

with the above config, i can see several PR created, this is exactly what I want.

**Reply by viceice** [reply]
Remove the package rule,it makes nothing.

https://docs.renovatebot.com/configuration-options/#packagerules

**Reply by codewode** [reply]
I got the example config from
https://github.com/lauromueller/unsocial-app/blob/master/renovate.json
https://www.youtube.com/watch?v=glNn6yFyl5U

after I followed your suggestion, the error is gone.

**Reply by rarkins** [reply]
Didn't you miss a line? https://github.com/lauromueller/unsocial-app/blob/ebcbdaa15c6e164176b9f457a82fc77e42a66b9c/renovate.json#L7

**Reply by codewode** [reply]
i didn't add on purpose, because I want to have a reviewer and tag the PR. The documentation is confusing, there are reviewers and assignees. Tried both with username, email. but it's not showing up. Here is my latest config

```
{
  "platform":"azure",
  "baseBranches":[
    "main"
  ],
  "assignees":[
    "xxx@xxx.xx"
  ],
  "reviewers":[
    "xxx@xxx.xx"
  ],
  "packageRules":[
    {
      "matchUpdateTypes":[
        "minor",
        "patch",
        "pin",
        "digest"
      ],
      "matchPackagePatterns":[
        "^(microsoft|Microsoft)"
      ],
      "groupName":"microsoft packages"
    }
  ]
}
```

what am I doing wrong and how do I add Tags to the PR?

**Reply by rarkins** [reply]
Be careful if you take somebody else's config and modify it without understanding what each piece does.

This discussion is solved and should not be continued now on a different topic. If you have questions about assignees and reviewers then please start a new discussion and describe it fully there.
